woocommerce/woocommerce-gateway-stripe
# SCA: Prepare a list of charges for subscriptions and fix what was broken in the meantime

Fixes #793

### Status

Because of an update on the way SCA treats saved cards during merchant-initiated payments, this PR might not be needed. Do not merge without further instructions.

### Changes proposed in this Pull Request:

- Add `load_previous_charges` to `WC_Stripe_Subs_Compat` in order to prepare the list that will be needed for subscriptions in a future Stripe API update.
- Add more metadata in `process_payment` in order to properly describe both the charge and the payment intent within Stripe (metadata & descriptions).

### Testing instructions

__Initial testing__

1. Activate WooCommerce subscriptions, create a subscription, and purchase it.
2. Put a breakpoint at the end of `load_previous_charges`.
3. Open the subscription in the *WooCommerce > Subscriptions* page in the dashboard.
4. Select the "Process Renewal" action in "Subscription Actions" and proceed.
5. You should see a list, containing the initial charge.

__Additional charges__

1. Remove the breakpoint and let a renewal proceed as it should.
2. Repeat steps 3-4 from the initial testing. The new charge should be included in the list.
